tianshou/data/batch.py

```diff
@@ -3,12 +3,25 @@
 from collections.abc import Collection
 from copy import deepcopy
 from numbers import Number
-from typing import Any, Dict, Iterable, Iterator, List, Optional, Sequence, Union
+from typing import (
+    Any,
+    Dict,
```

Comment:
I'd suggest using dict/list/tuple instead of Dict/List/Tuple everywhere

Reply:
the generics were supported in python 3.9, so `list[str]` and so on is not possible in 3.8. 

I also only use the builtin types in newer projects, but I guess you want to keep compatibility with 3.8, right?

Reply:
I very want to drop 3.8 tbh

Reply:
Let's do that in a separate PR, creating an issue for it now